Configuração de Bibliotecas de Validação e Manipuladores de Exceções de Resposta

Wfelipe2011 opened the pull request:
Esta solicitação de pull introduz várias melhorias relacionadas à configuração de autenticação e validação na aplicação.

- **Configuração de Bibliotecas de Validação e Manipuladores de Exceções:**
   - Configuradas bibliotecas de validação para lidar com a validação de dados em toda a aplicação.
   - Implementados manipuladores de exceções de resposta para gerar respostas apropriadas em caso de erros de validação.

- **Criação da UserModel e Contratos de Dados:** Foram adicionadas as definições da UserModel, juntamente com os contratos de dados necessários, tais como CreateUserParams, LoginParams, entre outros. Essa estruturação é fundamental para a gestão de usuários dentro do sistema.

- **Configuração do WhatsAppAdapter e Criação do WhatsAppService:** Incluímos a configuração inicial do WhatsAppAdapter e a implementação do WhatsAppService, permitindo a interação com a API do WhatsApp. Essa integração possibilita uma comunicação eficaz e direta com os usuários por meio dessa plataforma de mensagens.

- **Implementação de Endpoints de Autenticação:** Adicionamos os endpoints de Login, LoginCode, RecoverPassword (utilizando o WhatsAppService) e ResetPassword. Esses endpoints constituem o conjunto de autenticação do sistema, fornecendo funcionalidades cruciais para a segurança e gestão de contas dos usuários.

Essas melhorias visam fortalecer a segurança e confiabilidade da aplicação, garantindo que mecanismos adequados de autenticação e validação estejam em vigor.

Wfelipe2011 committed:
feat: configurar bibliotecas de validação e ajudantes de exceção de resposta

Configurou bibliotecas de validação e utilitários para lidar com a validação de dados. Implementou configuração para gerar respostas de exceção apropriadas em caso de erros de validação.

Wfelipe2011 committed:
feat: criar endpoint de login, middleware de autenticação e endpoint inicial para testar o middleware

Implementou o endpoint de login para permitir que os usuários façam login em suas contas. Criou middleware de autenticação para verificar se o usuário está autenticado em todas as rotas protegidas. Adicionou o endpoint inicial para testar o middleware de autenticação. Testou os endpoints e o middleware para garantir a funcionalidade correta.

Wfelipe2011 committed:
configura env de develop

Wfelipe2011 committed:
configurando serverless framework para usar env

Wfelipe2011 committed:
removendo url de hmg

Jwfelipee commented on src/contracts/CreateUserParams.ts:
```diff
@@ -0,0 +1,51 @@
+import { IsNotEmpty, IsOptional, IsString, Length } from "class-validator";
+import { BaseValidate } from "../shared/BaseValidate";
+
+type CreateUserParamsProps = {
+  name: string;
+  email: string;
+  cpf: string;
+  password: string;
+  avatar?: string;
+};
+
+export class CreateUserParams extends BaseValidate {
+  @Length(1, 255, { message: "A propriedade 'name' deve ter entre 1 e 255 caracteres" })
```
CreateUserParams? não deveria ser algo como CreateUser ou se isso for montar e validar os campos para criar um User, deveria ser InputCreateUser ao meu ver -> Input (entrada) Create (para criar) User (usuário) -> entrada para criar um usuário

Talvez o `Create` possa ser removido por ele ser o unico método a ser usado dessa classe, logo na hora de usar vai ser algo como -> `InputUser.create(...params)` e não `CreateUserParams.create(...params)`

Jwfelipee commented on src/contracts/CreateUserParams.ts:
```diff
@@ -0,0 +1,51 @@
+  @IsNotEmpty({ message: "A propriedade 'email' não pode ser vazia" })
+  email: string;
+
+  @Length(1, 11, { message: "A propriedade 'cpf' deve ter entre 1 e 11 caracteres" })
+  @IsString({ message: "A propriedade 'cpf' deve ser uma string" })
+  @IsNotEmpty({ message: "A propriedade 'cpf' não pode ser vazia" })
```
`@Length` já não garante que esse campo não vai ser vazio? é realmente necessário ter o `@IsNotEmpty`?

Isso vale para os outros campos também...

Wfelipe2011 committed:
feat: criar UserModel e contratos de dados

Adiciona a definição da UserModel e a criação de contratos de entrada de dados como CreateUserParams, LoginParams, etc.

Wfelipe2011 committed:
feat: configurar WhatsAppAdapter e criar WhatsAppService

Adiciona a configuração inicial do WhatsAppAdapter e a implementação do WhatsAppService para interagir com a API do WhatsApp.

Wfelipe2011 committed:
feat: implementar endpoints de autenticação

Adiciona os endpoints de Login, LoginCode, RecoverPassword (usando WhatsAppService) e ResetPassword, que compõem o conjunto de autenticação.

Wfelipe2011 committed:
fix: remover @IsNotEmpty redundante

Remove o decorador @IsNotEmpty redundante de campos onde não é necessário, simplificando as validações de entrada.

Wfelipe2011 committed:
refactor: ajustar nomes das classes de validação para Input<recurso>

Renomeia as classes de validação para seguir um padrão mais consistente, onde cada classe é nomeada como Input<recurso>, melhorando a organização e clareza do código.
